server/s3server.cc

```diff
@@ -96,6 +97,20 @@ std::set<struct s3_motr_idx_op_context *> global_motr_idx_ops_list;
 std::set<struct s3_motr_idx_context *> global_motr_idx;
 std::set<struct s3_motr_obj_context *> global_motr_obj;
 
+void s3_motr_init_timeout_cb(evutil_socket_t fd, short event, void *arg) {
+  s3_log(S3_LOG_ERROR, "", "Entering\n");
+  s3_iem(LOG_ALERT, S3_IEM_MOTR_CONN_FAIL, S3_IEM_MOTR_CONN_FAIL_STR,
+         S3_IEM_MOTR_CONN_FAIL_JSON);
+  event_base_loopbreak(global_evbase_handle);
+  return;
+}
+
+void *base_loop_thread(void *arg) {
+  pthread_detach(pthread_self());
+  event_base_loop(global_evbase_handle, EVLOOP_NO_EXIT_ON_EMPTY);
+  return NULL;
```

Comment:
Don't we need **conditional** exit() call here to terminate S3server process when event that is set to occur after 15secs is raised.
Presently, this thread will just terminate at line #111, but it won't terminate the other thread (in the process) where motr initliazation is carried out and stuck due to initialization issue in motr. This won't bring down the entire s3server process. Will it?

Reply:
Per discussion with Rajesh, if hax(part of hare), after receiving the IEM message from s3server indicating hang in motr initialization, restarts the entire IO stack (i.e motr services + s3service), then it is ok. Otherwise, these changes won't terminate S3server service as such, when there is motr initialization issue.